File: backend/Perflow/Domain/Playlist.cs
using Perflow.Domain.Abstract;
using Perflow.Domain.Enums;
using System.Collections.Generic;

namespace Perflow.Domain
{
    public sealed class Playlist : AuditEntity
    {
        public string Name { get; set; }

        public string Description { get; set; }

        public string IconURL { get; set; }

        public int AuthorId { get; set; }

        public User Author { get; set; }

        public AccessType AccessType { get; set; }

        public PlaylistType Type { get; set; }

        public ICollection<PlaylistSong> Songs { get; set; } 

        public ICollection<PlaylistReaction> Reactions { get; set; }

    }
}

File: frontend/src/app/models/playlist/playlist.ts
import { AccessType } from './accessType';
import { User } from '../user/user';
import { Song } from '../song/song';
import { PlaylistType } from '../enums/playlist-type';

export interface Playlist {
  id: number;
  createdAt: Date;
  name: string;
  description: string;
  iconURL: string;
  author: User;
  accessType: AccessType;
  type: PlaylistType;
  songs: Song[];
  isLiked: boolean;
}
